### DISCUSSION TITLE
A few collected questions from a new user

### DISCUSSION THREAD

**Discussion by JulianSMoore** [freeplane/freeplane#1690] [Help]
(I'm still checking out the presentation capabilities...)

May I pose a few related questions all together? I am trying to work out how to navigate and control presentation as part of preparing a presentation and all of these seem to belong together.

1. Floating Node - has no visible edge, so it appears _isolated_, but if I move it around it moves the rest of the map according to the layout rules. Is there a way to make it really float?

2. Navigate menu - "Unfold next presentation item" vs "Unfold next presentation item (fold)". The "(fold)" really baffles me - what do these do, and why is a "presentation" navigation option present when I'm not on the Presentations tab of the Panel?
 
3. Similar issue with "Go to next node (fold)", which actually _seems_ to unfold. And now I am more confused that ctrl-shift left/right arrows do not do/undo: I think my confusion is dues to not understanding what "next"/"previous" mean, e.g. "previous" does not mean "last selected", it seems to mean "predecessor", and "next" is "successor" in a depth first tree traversal. Is there a "goto next sibling"? or a way to traverse the tree breadth first? Have I missed the explanations of all this in the documentation somehow? If so, please steer me to the right place.

4. When folding/unfolding, is there any way to fold such that the layout does not change, i.e. node placement is unchanged as though the folded nodes were present but invisible? I find, for presentation purposes (or do I? I'm still finding my feet!), that when I fold to keep things simple, the layout is not aesthetically pleasing, but if I change it, the unfolded placements aren't right. Suggestions?

5. Trying to wrangle the layout, I notice that vertical spacing between siblings can be increased by dragging down a node, but if I later want to close the vertical gap between a node and its sibling vertically above, I can only do it with the lower sibling -- if I drag the upper sibling down it preserves empty space. I think there might be a good reason for that, but I can't see what it is. Can someone explain the rationale for this behaviour?

Many thanks!

Julian

**Comment by dpolivaev** [comment]
Please read this: https://chat.openai.com/share/d76d667d-71a7-41d8-8ce1-cc783d5a7ceb

About your question 4, I recommend using the "natural flow" layout because it keeps position of node being folded/unfolded and all its ancestor nodes.

**Reply by JulianSMoore** [reply]
OK, I'll try that. NB google site:docs.freeplane.org "natural flow" returns nothing, and searching the built in mindmap documentation didn't show that either. I remember seeing it as a label on the layout controls though... second row from the bottom, 4th from the left.

![image](https://github.com/freeplane/freeplane/assets/9352569/3d859cc3-a86d-4234-95e0-a841d2c72dfa)

but I couldn't work out how this worked: sometimes there are layouts like this

![image](https://github.com/freeplane/freeplane/assets/9352569/4413524c-b2da-40f4-bba5-90c31de5d361)

but I couldn't work out how to manage LHS and RHS separately.

If you can direct me to further help on that it would be much appreciated.

**Reply by dpolivaev** [reply]
Because I am a solo developer working on this project in my free time, the docu is sometimes not written unless someone from the community does it. You can get support in this forum and I would really appreciate it if you wrote down your learnings so I can add it to the documentation.

**Reply by dpolivaev** [reply]
Generally, the layout is configured at node level or for each map style separately and not at the map level. It gives you high flexibility but makes the whole less intuitive.

**Reply by JulianSMoore** [reply]
**Well, I have to say it is a really impressive tool -- and I looked at many such mapping tools before selecting Freeplane. My congratulations  on a nice, solid piece of work. Kudos! I have managed software development and I can code a bit myself, so I have some appreciation of your achievement.**

**I will try to share what I learn, but as I said earlier, I'm in a bit of a bind bootstrapping myself 😄 But, if you can clarify the following, I *will* write a neat summary**

I just looked at the styles, and all styles have this layout as seen in the style editor:

![image](https://github.com/freeplane/freeplane/assets/9352569/d57193d7-4c19-49e7-93cc-dd9f29acee9d)

But the root node still has this:

![image](https://github.com/freeplane/freeplane/assets/9352569/fd2ff076-8407-4a93-808c-8b2d17af6368)

I selected all nodes using a regex on blank and then chose "natural flow" as the layout, but to the right of the root node the layout is presented as

![image](https://github.com/freeplane/freeplane/assets/9352569/77d0cb41-1fc8-47d3-8b6a-dcb0da7e8d20)

and to its left as

![image](https://github.com/freeplane/freeplane/assets/9352569/5494f3a5-503c-47d2-a7f0-361f37d418f8)

so I don't know what I did.

The layout drop down shows "natural flow" highlighted... so one of my questions is, what is the uncollapsed "menu title" with the black arrow in the middle intended to communicate?

My guess is: LHS shows the layout specified for the Root node, right hand side shows the layout for the selected non-root node.... except that, if that's true then is there a big, because if I change the root node layout, the LHS of the "menu title" summary on non-root nodes is unaffected, almost as though every node could be a root node and needs to know that special layout just in case?

Would I be right in thinking that changing the selection in this

![image](https://github.com/freeplane/freeplane/assets/9352569/6797ec68-0842-4642-826d-2347f9cd989e)

has no effect if the selected node is not a root node... or does it? Does it interact if a non-root node is made the "view root node" via this control: 
![image](https://github.com/freeplane/freeplane/assets/9352569/61305c18-c96c-482f-82a9-1eb40df331e7)

**Reply by dpolivaev** [reply]
If you see two icons separated by the black left-to-right arrow, it means that the layout depends on the side of the node relative to its parent node. The left part shows the generic layout version chosen by the user and the right part shows which layout is selected for it at the current node position.

**Reply by dpolivaev** [reply]
Such generic layouts can have advantages for configuring styles or when node is moved to another side.

**Reply by JulianSMoore** [reply]
OK, maybe that will fall into place when I've spent a bit more time with Freeplane; thanks.

**Reply by euu2021** [reply]
This explanation about automatic (aka smart) layouts may help a bit: https://github.com/freeplane/freeplane/discussions/892#discussioncomment-4687077

> I selected all nodes using a regex on blank

To select all nodes in the map, select the Root, unfold all nodes, and use `Navigate->Select visible branch`.
Also, using Styles us very useful. Learn about it at: https://docs.freeplane.org/user-documentation/styles.html?highlight=styles#styles

**Comment by euu2021** [comment]
About 1: Use `Edit->Free positioned node (set/reset)`

**Reply by JulianSMoore** [reply]
Oh. I now realise again the tiny square in the menu margin is a check box. (Could those boxes be made bigger so they look more like check boxes?) **Got it, thanks**. (NB the "(set/reset)" text seems redundant)

**Comment by JulianSMoore** [comment]
A freeplane chatGPT tutor? That's so cool. Unfortunately, having used all the AI chat agents extensively (including one "fine tuned" - ha! - on my own documents, done locally of course), and being somewhat better informed than most about how they work etc. I trust them about as far as I can throw them on non-mainstream topics. 

Let's see if we can take the chatGPT output and build on it.

>Floating Node: Floating nodes are designed to visually stand out from the rest of the map by not having a visible edge connecting them to other nodes. While the concept of a "true float" where the node is entirely independent of the map's layout rules is appealing, the provided documentation does not explicitly mention a way to achieve this beyond the standard behavior of floating nodes. In Freeplane, floating nodes can be moved around more freely than other nodes, but they still obey the general layout rules to some extent. This means there isn't a built-in feature to completely isolate the positioning of floating nodes from the layout dynamics of the entire map.

**Helpful human said (see above): Use Edit->Free-positioned node** 🙂 

**Summary:** It is important to distinguish between _floating_ nodes (Format...Style... Floating Node, with checkbox indicator), and _freely positioned_ nodes (Edit... Free positioned node (set/reset), also with a checkbox indicator). Floating and Free-positioning are independent properties.

Floating nodes are simply nodes that have no visible _edge_ connecting them to their parent; they continue to behave like children of the parent for layout purposes, even if the parent-child relationship is not visible. Since there is currently no way to directly "goto" the parent of a node, to find the parent you must select a style that has a visible edge to make the relationship visible. Re-applying the Floating Node style will re-hide the edge.

Free-positioned nodes can be moved around without affect the layout of other nodes, regardless of whether or not those other nodes are free positioned. If a node ceases to be free-positioned (select Edit... Free positioned node (set/reset); it will be assigned a position relative to its parent according to the layout rule applicable.

>Navigate Menu - Unfold Next Presentation Item vs. Fold: The "Unfold next presentation item" and "Unfold next presentation item (fold)" commands in the Navigate menu are designed for use during presentations. The documentation doesn't provide specific details on these commands, but typically:
>"Unfold next presentation item" would go to the next item in your presentation plan, unfolding any folded nodes to reveal them as part of the presentation.
>The "(fold)" variant might imply a behavior where it also automatically folds back previously unfolded items to focus attention on the current item. However, without explicit documentation, this is a bit speculative. 

**If chatGPT says it's speculating who am I to disagree! If the documentation doesn't speak to this, then it's using "natural language" to guess at a meaning.** I still don't understand this.

>The presence of presentation navigation options outside of presentation mode could be for convenience, allowing users to prepare or review their presentation paths without being in the full presentation view.

**I thought it might be a convenience of some sort, but I haven't yet worked it out.**

>Go to Next Node (fold) and Navigation Confusion: In tree structures like those in Freeplane, "next" and "previous" typically refer to traversal order, often depth-first as you've surmised. This means "next" would move you to the successor in a depth-first search, and "previous" to the predecessor. There isn't specific mention in the provided documentation about a "goto next sibling" feature or a way to traverse the tree in a breadth-first manner, which suggests these might not be directly supported in the way you're looking for.

**AFAICT ctrl+alt+ left/right arrow is going next/previous in a depth first traversal. Up/down arrows go towards the beginning/end of the current level.. except that at the bottom (on the last node of that level) further down arrow movement starts scrolling down, and at the top, well, I don't know: I have some floating nodes, and up from the top of any level (1, 2, 3...) UP takes me to a floating node, whether that node is above the current selection or not (according to whether or not it is free positioned, but still in a parent-child relationship)**

>Folding/Unfolding and Layout Changes: The behavior you're observing with folding/unfolding affecting layout aesthetics is inherent to how mind mapping software typically manages space and organization. The documentation does not specifically address methods to fold nodes without changing the visual layout, as the software automatically adjusts the layout to optimize space and visibility. For presentation purposes, achieving a balance between simplicity and aesthetic appeal might require manual adjustment or exploring different layout styles and formats provided by the software.`

**This is standard chatGPT: state the obvious/implied.** I will look to see how making everything free-positioned might be used to do esp unfolding without affecting things: it sounds as though it should work.

>Vertical Spacing and Node Dragging: The rationale behind being able to increase vertical spacing by dragging down a node but not being able to easily close gaps by moving an upper node downwards could be related to preserving the clarity and visual structure of the mind map. This behavior ensures that manually created spacing (for emphasis or organization) is not unintentionally lost by upward movements. The documentation does not explicitly justify this design choice, but it aligns with common practices in mind mapping and diagramming tools to maintain user-defined spacing and layout preferences.

**That's basically just a reformulation/description of the observed behaviour, not an explanation as such, but it's not a big deal.**

>For detailed exploration of these features and potentially finding more direct answers or workarounds, I recommend looking into the Freeplane manual or help resources directly within the software, as the provided documentation does not cover these specific queries in detail.

**LOL! The amount of time I have spent crafting (usually unsuccessfully) prompts that force AI agents just to say "IDK!" and then shut up** 😆

**Reply by dpolivaev** [reply]
The Freeplane tutor GPT can read documents for you and try to answer your questions based on the given information. I gave it the content of https://docs.freeplane.org/ and the documentation mind map. It can save your time, but it is on you to check if the suggestions make sense. But the documents are obsolete at the moment of writing and they age very fast.

**Reply by JulianSMoore** [reply]
@dpolivaev Selecting all nodes and marking them as free positioned collapses them all into a heap on the root 😄 And now I look again at a single node I should have expected that! My bad.

For future reference, I guess the nodes have a *layout* position that isn't stored with the node itself...? If the position calculated were stored with the node then setting free-position should leave it where it was until moved. That would be very handy.

**Reply by dpolivaev** [reply]
Some things are harder to implement than some other things. And in case you do not like the result of an action you can hit undo.

**Reply by JulianSMoore** [reply]
When I click the link to the Freeplane tutor GPT, I can't ask questions unless I have a GPT plus subscription (which I don't).

I don't think I did anything wrong, but don't worry about it, just wanted to record the observation.

**Reply by dpolivaev** [reply]
Yes, it is how open AI made it. I use chat GPT so much and for so many purposes that I do not want to miss the subscription.
